kasa/cli.py

```diff
@@ -513,12 +513,14 @@ async def sysinfo(dev):
 
 @cli.command()
 @pass_dev
-async def state(dev: SmartDevice):
+@click.pass_context
+async def state(ctx, dev: SmartDevice):
     """Print out device state and versions."""
+    verbose = ctx.parent.params.get("verbose", False) if ctx.parent else False
+
```

Comment:
Likewise, it'd be better to move the functional changes into their own PR(s).

Reply:
Two of the uncovered pieces of code were simply properties of the TapoDevice, i.e. features and device_id. This change brought in some code to really use those so they could be put into a sensible test.